projects/cashmere/src/lib/sass/stepper.scss

```diff
@@ -0,0 +1,311 @@
+$track-space: 4rem;
+$isolated-disk-size: 2rem;
+
+@mixin disk-border($border-color) {
+    box-shadow: 0px 0px 0px 3px $border-color;
+}
+
+@mixin step-arrow($top-bottom, $right) {
+    @if $right {
+        border-right: $track-space * 0.05 solid $right;
+    }
+    border-left: $track-space / 4 solid transparent;
+    border-top: $track-space / 2 solid $top-bottom;
+    border-bottom: $track-space / 2 solid $top-bottom;
+    display: block;
+    height: 0px;
+    position: absolute;
+    right: 0;
+    top: 0;
+    z-index: 2;
+}
+
+@mixin arrow-theme($active, $complete, $track-bg, $border, $disk-bg, $text) {
+    .hc-stepper-track {
+        background-color: $track-bg;
+    }
+
+    .hc-step {
+        border-top-color: $track-bg;
+
+        &.hc-step-active {
+            background-color: $active;
+
+            &:not(.hc-step-last) {
+                .hc-step-arrow {
+                    @include step-arrow($track-bg, null);
+                }
+            }
+
+            .hc-step-disk {
```

Comment:
A lot of this CSS isn't up to standard, per step 6 of https://cashmere.healthcatalyst.net/guides/using-customizing-components.

Quoted:

- The mixin stylesheet should not include any nested CSS selectors that would make confusing assumptions about the user's HTML structure. That is, the [disabled] attribute selector and pseudo-selectors like :hover or :after are okay, but class or tag names like .hc-checked and input should be avoided.
- Both stylesheets should be as flat as possible. Nesting CSS selectors in SCSS files creates extremely specific selectors, which are hard to override. Cashmere should make it easy to override a style rule if needed. CSS classes can be added to component markup to avoid deep nesting.

Reply:
@isaaclyman I've got all the base classes following the standard (I think). But the big exception are the mixins to theme the arrow or isolated types: `arrow-theme()` and `isolated-theme()`.  Because the point of those functions is to override the base classes with specific color values - I'm really struggling to see how to do that without nested selectors.  

Part of the problem is that within those those mixins - the trailing & operator doesn't work.  Because the browser assumes the parent to be the class that called the mixin - not the class in the mixin.  So the hierarchy gets pretty wonky.

I could flatten everything out to not be nested but use > selectors - but it seems like that's just semantics, and it would actually make the hierarchy assumptions even worse. 

Open to suggestions...

Reply:
I take it back - it was doable, just required a good bit of rework.  It makes the css a little crazy, but I think now everything is as flat as it can be.  Give it a look and let me know what you think.